### PR TITLE
[20603] Upgrade dependency version to last patch version in .repos file

### PR DESCRIPTION
**Pull request by elianalf**
## Description

This PR changes dependency version to last patch in .repos file

## Contributor Checklist
- N/A Commit messages follow the project guidelines. 
- N/A The code follows the style guidelines of this project. 
- N/A Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- N/A Any new/modified methods have been properly documented using Doxygen. 
- N/A Changes are ABI compatible. 
- N/A Changes are API compatible. 
- N/A New feature has been added to the `versions.md` file (if applicable).
- N/A New feature has been documented/Current behavior is correctly described in the documentation. 
    
- N/A Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by JesusPoderoso — APPROVED** [review]

**Review by JesusPoderoso — DISMISSED** [review]

File: fastrtps.repos [hunk 1]
```diff
@@ -6,16 +6,16 @@ repositories:
...
     fastddsgen:
         type: git
         url: https://github.com/eProsima/Fast-DDS-Gen.git
-        version: v2.1.2
+        version: 2.1.x
     fastddsgen/thirdparty/idl-parser:
         type: git
         url: https://github.com/eProsima/IDL-Parser.git
-        version: v1.2.0
+        version: 1.2.x
```
Comment (Mario-DL) [inline fastrtps.repos:21]: Im afraid that neither `fastddsgen` nor `idl-parser` have those branches. We can leave it with the fixed tags

**Review by Mario-DL — CHANGES_REQUESTED** [review]

File: fastrtps.repos (hunk 1 above)
Comment (Mario-DL) [reply fastrtps.repos:21]: As we agreed, we would need to create those branches in the repos

Comment (MiguelCompany) [reply fastrtps.repos:21]: Note: We agreed to remove idl-parser from the .repos file, and create 2.1.x in Fast-DDS-Gen

Comment (Mario-DL) [reply fastrtps.repos:21]: Exactly, I forgot updating my previous comment after making decision on what to do with `idl-parser`. Thanks !

**Review by Mario-DL — APPROVED** [review]

**Pull request merged**